### DISCUSSION TITLE
Save file picker is not returns selected file type

### DISCUSSION THREAD

**Discussion by bronnikovgs** [AvaloniaUI/Avalonia#8978] [Q&A] [answered]
i'm use Avalonia 11 preview1 and trying to save file with SaveFilePicker. How can I get the file type selected from the FileTypeChoices list in Windows and Linux?

**Comment by maxkatz6** [comment]
We don't have such API right now, but it can be added to the IStorageFile interface with no issue.
The problem is to how to know about file type. 
Do you expect file's content type, mime type? If so, not every backend/OS even has this information. So I guess it would be nullable in the API.
Extension type can be easily get from the file name. But that's not really reliable way to get file type.

**Comment by bronnikovgs** [comment]
SaveFilePickerAsync could return a structure consisting of IStorageFile and FilePickerFileType. Is this option possible?

**Reply by maxkatz6** [reply]
FileType can be part of IStorageFile. But the problem is OS support.

**Comment by bronnikovgs** [comment]
At the time of pressing the OK button in the Save File Picker, the file itself is not yet there, but we have forcibly specified the file type (if it is not *.*). Is it possible to get at least the selected index in the File Type Choices array?

**Reply by maxkatz6** [reply]
Depends on the OS. On win and linux it's possible, but we know only selected type there, not the actual content type.
On android and wasm we have only mime type of the file, which might be different from the file extension. 
On mac and ios we don't have any information as far as I remember. 

But if you need a file content type, probably the best way is to use mine guess tools/libraries.

**Comment by bronnikovgs** [comment, marked as answer]
OK, thanks!
